staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -876,12 +911,13 @@ func (p *sharedProcessor) distribute(obj interface{}, sync bool) {
 }
 
 func (p *sharedProcessor) run(ctx context.Context) {
+	logger := klog.FromContext(ctx)
 	func() {
 		p.listenersLock.RLock()
 		defer p.listenersLock.RUnlock()
 		for listener := range p.listeners {
-			p.wg.StartWithContext(ctx, listener.run)
-			p.wg.StartWithContext(ctx, listener.pop)
+			p.wg.Start(func() { listener.run(logger) })
+			p.wg.Start(func() { listener.pop(logger) })
```

Comment:
This is not entirely correct: the listener should log panics with the logger that it was created with, not the one of the informer. That becomes relevant when `WithName` is used to identify output from different event handlers.

Will add tests and fix this. I didn't want to store a logger in the listener (conceptually similar to storing a context in a struct), but it seems like the right approach here.

Reply:
Done. Keeping a separate fixup commit got confusing, so I applied them to clean up the PR.